functions/assignOfflineContact.ts

```diff
@@ -0,0 +1,141 @@
+import '@twilio-labs/serverless-runtime-types';
+import {
+  Context,
+  ServerlessCallback,
+  ServerlessFunctionSignature,
+} from '@twilio-labs/serverless-runtime-types/types';
+import {
+  responseWithCors,
+  bindResolve,
+  error400,
+  error500,
+  success,
+  send,
+} from '@tech-matters/serverless-helpers';
+// eslint-disable-next-line prettier/prettier
+import type { PromiseValue } from 'type-fest';
+
+const TokenValidator = require('twilio-flex-token-validator').functionValidator;
+
+type EnvVars = {
+  TWILIO_WORKSPACE_SID: string;
+  TWILIO_CHAT_TRANSFER_WORKFLOW_SID: string;
+};
+
+export type Body = {
+  targetSid?: string;
+  finalTaskAttributes: string;
+};
+
+// eslint-disable-next-line prettier/prettier
+type TaskInstance = PromiseValue<ReturnType<ReturnType<ReturnType<ReturnType<Context['getTwilioClient']>['taskrouter']['workspaces']>['tasks']>['fetch']>>;
+// eslint-disable-next-line prettier/prettier
+type WorkerInstance = PromiseValue<ReturnType<ReturnType<ReturnType<ReturnType<Context['getTwilioClient']>['taskrouter']['workspaces']>['workers']>['fetch']>>;
+
+const assignToAvailableWorker = async (
+  event: Body,
+  newTask: TaskInstance,
+) => {
+  const reservations = await newTask.reservations().list();
+  const reservation = reservations.find(r => r.workerSid === event.targetSid);
```

Comment:
What if the worker got a real transfer at the same time?

Reply:
This are the reservation for the newly created task only, so if there are other tasks around, this should not interfere. Also given that we are using default channel, this kind of task will skip the "waiting queue" even if the counselor is at max capacity.

Reply:
Oh right!  I missed that this is just reservations on the `newTask`